debian/build.sh

```diff
@@ -0,0 +1,194 @@
+                        OUR_ARCH=${DEB_ARCH}
+                        ;;
+                    arm)
+                        DEB_ARCH=armhf
+                        OUR_ARCH=${1}
+                        ;;
+                    ppc64le)
+                        DEB_ARCH=ppc64el
+                        OUR_ARCH=${1}
+                        ;;
+                    s390)
+                        DEB_ARCH=${1}x
+                        OUR_ARCH=${1}
+                        ;;
+                    x86_64)
+                        DEB_ARCH=amd64
+                        OUR_ARCH=${1}
+                        ;;
+                    *) die "${1} is not supported by this script!" ;;
+                esac
+                ;;
+            -h | --help)
+                echo
+                cat "${DEBIAN}"/README.txt
+                echo
+                exit 0
+                ;;
+            -l | --ltp)
+                LTP=true
+                ;;
+            -m | --modules-folder)
+                shift
+                MODULES_FOLDER=${1}
+                [[ -d ${MODULES_FOLDER} ]] || die "${MODULES_FOLDER} specified but it does not exist!"
+                ;;
+            -p | --password)
+                shift
+                DEB_PASS=${1}
+                ;;
+            -u | --user)
+                shift
+                DEB_USER=${1}
+                ;;
+        esac
+        shift
+    done
+}
+
+# Checks if command is available
+function is_available() {
+    command -v "${1}" &>/dev/null || die "${1} needs to be installed!"
+}
+
+# Do some initial checks for environment and configuration
+function reality_checks() {
+    # Validity checks
+    [[ -z ${DEB_ARCH} ]] && die "'-a' is required but not specified!"
+
+    # Some tools are in /usr/sbin or /sbin but they might not be in PATH by default
+    [[ ${PATH} =~ /sbin ]] || PATH=${PATH}:/usr/sbin:/sbin
+    is_available blkid
+    is_available findmnt
+    is_available mkfs.ext4
+    is_available qemu-debootstrap
+    is_available qemu-img
+
+    # Default values
+    [[ -z ${DEB_USER} ]] && DEB_USER=user
+    [[ -z ${DEB_PASS} ]] && DEB_PASS=password
+    [[ -z ${LTP} ]] && LTP=false
+}
+
+# Build image
+function create_img() {
+    WORK_DIR=$(mktemp -d -p "${DEBIAN}")
+    ORIG_USER=$(logname)
+
+    set -x
+
+    # Create the image that we will use and mount it
+    IMG=${WORK_DIR}/debian.img
+    qemu-img create "${IMG}" 5g
+    mkfs.ext4 "${IMG}"
+    MOUNT_DIR=${WORK_DIR}/rootfs
+    mkdir -p "${MOUNT_DIR}"
+    mount -o loop "${IMG}" "${MOUNT_DIR}"
+
+    # Install packages
+    PACKAGES=(
+        autoconf
+        automake
+        bash
+        bison
+        build-essential
+        ca-certificates
+        flex
+        git
+        libtool
+        m4
+        pkg-config
+        stress-ng
+        sudo
+        vim
+    )
+    qemu-debootstrap --arch "${DEB_ARCH}" --include="${PACKAGES[*]//${IFS:0:1}/,}" buster "${MOUNT_DIR}" || exit ${?}
```

Comment:
do we want to move buster up to a global var near the top of the file, that way it's easier to upgrade to newer releases of debian some day?

Reply:
Yes and I could turn it into a parameter as well (`-v`/`--version` maybe?)